hibernate-reactive-core/src/main/java/org/hibernate/reactive/loader/ast/internal/ReactiveCompoundNaturalIdLoader.java

```diff
@@ -0,0 +1,265 @@
+            fetches.add( fetch );
+        }
+        return fetches.build();
+    }
+
+    protected CompletionStage<T> reactiveSelectByNaturalId(
+            Object bindValue,
+            NaturalIdLoadOptions options,
+            BiFunction<TableGroup, LoaderSqlAstCreationState, DomainResult<?>> domainResultProducer,
+            LoaderSqlAstCreationState.FetchProcessor fetchProcessor,
+            Function<Boolean, Long> statementStartHandler,
+            BiConsumer<Object, Long> statementCompletionHandler,
+            SharedSessionContractImplementor session) {
+        {
+            final SessionFactoryImplementor sessionFactory = session.getFactory();
+            final JdbcServices jdbcServices = sessionFactory.getJdbcServices();
+            final JdbcEnvironment jdbcEnvironment = jdbcServices.getJdbcEnvironment();
+            final SqlAstTranslatorFactory sqlAstTranslatorFactory = jdbcEnvironment.getSqlAstTranslatorFactory();
+
+            final LockOptions lockOptions;
+            if (options.getLockOptions() != null) {
+                lockOptions = options.getLockOptions();
+            } else {
+                lockOptions = LockOptions.NONE;
+            }
+
+            final NavigablePath entityPath = new NavigablePath(entityDescriptor().getRootPathName());
+            final QuerySpec rootQuerySpec = new QuerySpec(true);
+
+            final LoaderSqlAstCreationState sqlAstCreationState = new LoaderSqlAstCreationState(
+                    rootQuerySpec,
+                    new SqlAliasBaseManager(),
+                    new SimpleFromClauseAccessImpl(),
+                    lockOptions,
+                    fetchProcessor,
+                    true,
+                    sessionFactory
+            );
+
+            final TableGroup rootTableGroup = entityDescriptor().createRootTableGroup(
+                    true,
+                    entityPath,
+                    null,
+                    () -> rootQuerySpec::applyPredicate,
+                    sqlAstCreationState,
+                    sessionFactory
+            );
+
+            rootQuerySpec.getFromClause().addRoot(rootTableGroup);
+            sqlAstCreationState.getFromClauseAccess().registerTableGroup(entityPath, rootTableGroup);
+
+            final DomainResult<?> domainResult = domainResultProducer.apply(rootTableGroup, sqlAstCreationState);
+
+            final SelectStatement sqlSelect = new SelectStatement(rootQuerySpec, Collections.singletonList(domainResult));
+
+            final JdbcParameterBindings jdbcParamBindings = new JdbcParameterBindingsImpl(naturalIdMapping().getJdbcTypeCount());
+
+            // TODO:  Uncomment and FIX
+//            applyNaturalIdRestriction(
+//                    bindValue,
+//                    rootTableGroup,
+//                    rootQuerySpec::applyPredicate,
+//                    jdbcParamBindings::addBinding,
+//                    sqlAstCreationState,
+//                    session
+//            );
```

Comment:
Why did you comment this?

Reply:
Having trouble uncommenting it and fixing the compile errors

rootQuerySpec::applyPredicate results in:  `Incompatible types: Object is not convertible to Predicate`
jdbcParamBindings::addBinding :  Incompatible types: Object is not convertible to JdbcParameter

Reply:
Ok, I will have a look